src/components/base/RegisterForm.tsx

```diff
@@ -0,0 +1,118 @@
+import InputField from './Input'
+import Button from './Button'
+import { useState } from 'react'
+import { Credentials } from '../../entities/Credentials'
+
+export type InputFieldProps = {
+    username?: string
+    password?: string
+
+    repeatPassword?: string
+    name?: string
+    firstname?: string
+    email?: string
+    telephone?: string
+    street?: string
+    place?: string
+    zip?: string
+}
+
+const RegisterForm = (_props: InputFieldProps) => {
+    const [credentials, setCredentials] = useState<Credentials>({
```

Comment:
You probably don't need state here, you can use the props values, and you should pass updates directly to the parent via a function like `onChange(credentials: Credentials)`. You should add this function to the props.

Reply:
moved useState to RegisterView